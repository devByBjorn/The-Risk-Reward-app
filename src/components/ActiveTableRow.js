import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
// import { calculatePositiveR, calculateNegativeR } from '../calculations/riskRewardCalculation'
import { EditIcon, TrashIcon } from '../icons/IconsComponents'
import { deleteFirebaseTrade } from '../actions/tradeActions'

const ActiveTableRow = (
  {
    dispatch,
    market,
    direction,
    setup,
    entry,
    stop,
    target,
    status,
    opened,
    negativeR,
    positiveR,
    id }) => (
    status === 'active' &&
    <tr>
      <td>{market}</td>
      <td>{direction}</td>
      <td>{setup}</td>
      <td>{entry}</td>
      <td>{stop}</td>
      <td>{target}</td>
      <td>{opened}</td>
      <td>{negativeR}</td> {/* loss -R*/}
      <td>{positiveR}</td> {/* win +R*/}
      <td>
        <div>
          <Link to={`/edit-trade/${id}`}>
            <EditIcon />
          </Link>
          <button onClick={() => {
            dispatch(deleteFirebaseTrade({ id }))
          }}
          ><TrashIcon />
          </button>
        </div>
      </td>
    </tr>
  )

export default connect()(ActiveTableRow)