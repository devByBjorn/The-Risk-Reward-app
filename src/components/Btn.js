import React from 'react'

export const Btn = (props) => (
  <button
    type="button"
    {...props}
  >
    {props.text}</button>
)

export const SubmitBtn = (props) => (
  <button>{props.text}</button>
)

export default Btn