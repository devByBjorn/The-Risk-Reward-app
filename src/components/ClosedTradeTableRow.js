import React, { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { EditIcon, TrashIcon } from '../icons/IconsComponents'
import { deleteTrade } from '../actions/tradeActions'




const ClosedTradeTableRow = (
  {
    dispatch,
    market,
    direction,
    setup,
    opened,
    closed,
    status,
    outcome,
    rewardToRisk,
    id }) => {

  const [rowColor, setRowColor] = useState('')

  const colors = {
    win: '#7FBA00',
    loss: '#F0341B',
    scratch: '#FFB900'
  }
  const { win, loss, scratch } = colors

  useEffect(() => {
    if (outcome === 'win') {
      setRowColor(win)
    } else if (outcome === 'loss') {
      setRowColor(loss)
    } else {
      setRowColor(scratch)
    }
  })

  return (
    status === 'closed' &&

    <tr style={{ backgroundColor: rowColor }}>
      <td>{market}</td>
      <td>{direction}</td>
      <td>{setup}</td>
      <td>{rewardToRisk}</td>
      <td className="desktop">{opened}</td>
      <td className="desktop">{closed}</td>
      <td className="desktop">{outcome}</td>
      <td className="handle-trade-row">
        <div className="handle-trade">
          <Link to={`/edit-trade/${id}`}>
            <EditIcon
              className="edit-icon icon"
            />
          </Link>
          <button
            className="btn-transparent"
            onClick={() => {
              alert('Do you want to delete this trade?')
              dispatch(deleteTrade({ id }))
            }}
          ><TrashIcon
              className="trashcan-icon icon"
            />
          </button>
        </div>
      </td>
    </tr>
  )
}
export default connect()(ClosedTradeTableRow)