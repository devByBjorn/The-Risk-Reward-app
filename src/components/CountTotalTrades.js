import React from 'react'
import { connect } from 'react-redux'

const ClosedTradesTotal = ({ trades }) => (
  <h2>Total Closed Trades: {trades.length}</h2>
)

const mapStateToProps = (state) => {
  return {
    trades: state.trades.filter((trade) => trade.status === 'closed')
  }
}

export default connect(mapStateToProps)(ClosedTradesTotal)