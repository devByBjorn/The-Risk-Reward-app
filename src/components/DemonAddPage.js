import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'
import { addDemon } from '../actions/demonActions'
import DemonFinderForm from './demon-finder/DemonFinderForm'

const AddDemonPage = (props) => (
  <div>
    <h1>Add Demon Page</h1>
    <NavLink to="/demons">Back to demon list</NavLink>
    <div>
      <ol>
        <li>Give your demon suiting a title</li>
        <li>Think about your demon in terms of what,when,why,  how.
          <ul>
            <li>What makes the demon take over?</li>
            <li>When does this demon take over?</li>
            <li>Why...</li>
            <li>How can you manage this demon?</li>
          </ul>
        </li>
        <li>Write down your thoughts and process of managing this demon</li>
      </ol>
      <div>
        <h2>Demon</h2>
        <DemonFinderForm
          handleSubmit={(demon) => {
            props.dispatch(addDemon(demon))
            props.history.push('/demons')
          }}
        />
      </div>
    </div>
  </div>
)

export default connect()(AddDemonPage)