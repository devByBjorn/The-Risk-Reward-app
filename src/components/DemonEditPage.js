import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'
import DemonFinderForm from '../components/demon-finder/DemonFinderForm'
import { editDemon, deleteDemon } from '../actions/demonActions'
import { TrashIcon } from './table/TradeTableIcons'

class EditDemonPage extends React.Component {
  handleAddEdit = (demon) => {
    const { id } = this.props.demon
    this.props.editDemon(id, demon)
    this.props.history.push('/demons')
  }
  handleDelete = () => {
    const { id } = this.props.demon
    this.props.deleteDemon({ id })
    this.props.history.push('/demons')
  }
  render() {
    return (
      <div>
        <h1>Edit/Read Demon Page</h1>
        <NavLink to="/demons">Back to demon list</NavLink>
        <DemonFinderForm
          demon={this.props.demon}
          handleSubmit={this.handleAddEdit}
        />
        <button
          onClick={this.handleDelete}
        ><TrashIcon />
        </button>
      </div>
    )
  }
}


const mapStateToProps = (state, props) => ({
  demon: state.demons.find((demon) => demon.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch) => ({
  editDemon: (id, demon) => dispatch(editDemon(id, demon)),
  deleteDemon: (id) => dispatch(deleteDemon(id))
})


export default connect(mapStateToProps, mapDispatchToProps)(EditDemonPage)