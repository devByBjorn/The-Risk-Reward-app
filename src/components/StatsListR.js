import React from 'react'
import { connect } from 'react-redux'
import { getTotal, getAvarage, getLowest, getHighest } from '../calculations/calcR'
import HighlightedList from '../components_style/HighlightedList'
import ListItem from '../components_style/ListItem'
import Span from '../components_style/Span'
import SubHeading from '../components_style/SubHeadingStyled'
import { StatsListWrapper } from '../components_style/StatsListWrapperStyled'

export const StatsListR = ({ trades }) => {
  const worstR = getLowest(trades, 'rMultiple')
  const bestR = getHighest(trades, 'rMultiple')
  const totalR = getTotal(trades, 'rMultiple')
  const averageR = getAvarage(trades, 'rMultiple')

  const listItems = [totalR, bestR, worstR, averageR]
  const itemsHeadings = ['Total', 'Best', 'Worst', 'Average']

  return (
    <StatsListWrapper flexDirection="column">
      <SubHeading>R</SubHeading>
      <HighlightedList>
        {listItems.map((item, i) => (
          <ListItem key={item}>
            <Span>{itemsHeadings[i]}</Span>
            {trades.length ? item : '-'}
          </ListItem>
        ))}
      </HighlightedList>
    </StatsListWrapper>
  )
}

const mapStateToProps = (state) => ({
  trades: state.trades.filter((trade) => trade.status === 'closed')
})


export default connect(mapStateToProps)(StatsListR)