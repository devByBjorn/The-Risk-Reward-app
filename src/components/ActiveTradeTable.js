import React from 'react'
import { connect } from 'react-redux'
import ActiveTableRow from './ActiveTableRow'
import { ThMobileAndDesktop, ThOnlyDesktop } from './TradeTableHead'
const ActiveTradeTable = ({ trades }) => (
  <div>
    <table className="table-grid active">
      <caption>Active Trades</caption>
      <thead>
        <tr>
          <ThMobileAndDesktop heading='Market' />
          <ThMobileAndDesktop heading='Direction' />
          <ThMobileAndDesktop heading='Setup' />
          <ThMobileAndDesktop heading='Entry' />
          <ThMobileAndDesktop heading='Stop' />
          <ThMobileAndDesktop heading='Target' />
          <ThOnlyDesktop heading='Opened' />
          <ThMobileAndDesktop heading='-R' />
          <ThMobileAndDesktop heading='+R' />
        </tr>
      </thead>
      <tbody>
        {trades.map((trade) =>
          (<ActiveTableRow
            {...trade}
            key={trade.id}
          />)
        )}
      </tbody>
    </table>
  </div>
)

const mapStateToProps = (state) => {
  return {
    trades: state.trades,
  }
}

export default connect(mapStateToProps)(ActiveTradeTable)