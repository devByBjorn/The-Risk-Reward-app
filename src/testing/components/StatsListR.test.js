import React from 'react'
import { shallow } from 'enzyme'
import { StatsListR } from '../../components/StatsListR'
import { trades } from '../fixtures/trades'

test('should render StatsListR', () => {
  const wrapper = shallow(
    <StatsListR
      trades={trades}
    />
  )
  expect(wrapper).toMatchSnapshot()
})

