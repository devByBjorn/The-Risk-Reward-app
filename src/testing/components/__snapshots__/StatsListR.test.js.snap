// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render StatsListR 1`] = `
<StatsListWrapperStyled__StatsListWrapper
  flexDirection="column"
>
  <SubHeadingStyled__SubHeading>
    R
  </SubHeadingStyled__SubHeading>
  <HighlightedList>
    <ListItem
      key="3"
    >
      <Span>
        Total
      </Span>
      3
    </ListItem>
    <ListItem
      key="2"
    >
      <Span>
        Best
      </Span>
      2
    </ListItem>
    <ListItem
      key="-1"
    >
      <Span>
        Worst
      </Span>
      -1
    </ListItem>
    <ListItem
      key="1"
    >
      <Span>
        Average
      </Span>
      1
    </ListItem>
  </HighlightedList>
</StatsListWrapperStyled__StatsListWrapper>
`;
